basics/trey-hunner-zip-and-enumerate.md

```diff
@@ -76,9 +76,110 @@ c 3
 >>>
 ```
 
-Now you're ready to read [this awesome looping
-tutorial](http://treyhunner.com/2016/04/how-to-loop-with-indexes-in-python/).
-Read it now, then come back here and do the exercises.
+## zip
+
+What comes to your mind when you hear the word `zip`? A mechanism extensively used to tie two parts of something i.e. Shirt, Jacket etc. The `zip()` functions does pretty much same, it helps us tie two or more different items together.
+
+```python
+>>> users = ["Tushar", "Aman", "Anurag", "Sohit"]
+>>> uids = ["usr122", "usr123", "usr124", "usr125"]
+>>> user_details = zip(uids, users)
+>>> print(user_details)
+<zip object at 0x0000016938E311C0>
+>>>
+```
+
+After calling zip, an iterator is returned. In order to see the content wrapped inside, we need to first convert it to a list. Basically, `zip()` is an iterator, i.e. lazy: it gives the items as needed, instead of calculating them and storing them into memory all at once. So the zip object cannot show its elements before the elements are used, because it hasn't computed them yet.
+
+```python
+>>> users = ["Tushar", "Aman", "Anurag", "Sohit"]
+>>> uids = ["usr122", "usr123", "usr124", "usr125"]
+>>> user_details = zip(uids, users)
+>>> print(list(user_details))
+[('usr122', 'Tushar'), ('usr123', 'Aman'), ('usr124', 'Anurag'), ('usr125', 'Sohit')]
+>>>
+```
+
+If the lists are of different lengths, some items from the end of the longer list will be ignored.
+
+```python
+>>> users = ["Tushar", "Aman", "Anurag"]
+>>> emails = ["tushar@example.com", "aman@example.com", "anurag@example.com", "sohit@example.com"]
+>>> users_contact = zip(users, emails)
+>>> print(list(users_contact))
+[('Tushar', 'tushar@example.com'), ('Aman', 'aman@example.com'), ('Anurag', 'anurag@example.com')]
+>>>
+```
+
+Here the shortest list is `users`, with length 3, so `zip(users, emails)` only takes the first 3 emails.
+> We do not recommend calling `zip()` with lists of different lengths.
```

Comment:
```suggestion
We do not recommend calling `zip()` with lists of different lengths, because ignoring items is usually not what you intended to do.
```